### PR TITLE
chore: retry on codecov failing error

### PR DESCRIPTION
**Pull request by ologbonowiwi**

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -80,7 +80,6 @@ jobs:
         with:
           token: ${{ secrets.CODECOV_TOKEN }} # not required for public repos
           files: lcov.info
-          fail_ci_if_error: true
```
Comment (meskill) [inline .github/workflows/ci.yml:83]: we could miss some important information without coverage and will end up with rerunning the job

we can try rerun it automatically like it's done [here](https://github.com/tagatac/bagoup/pull/57/files)

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -76,11 +76,15 @@ jobs:
         run: cargo llvm-cov --all-features --lcov --output-path lcov.info
 
       - name: Upload Coverage to Codecov
-        uses: codecov/codecov-action@v4
+        uses: Wandalen/wretry.action@v1
         with:
-          token: ${{ secrets.CODECOV_TOKEN }} # not required for public repos
-          files: lcov.info
-          fail_ci_if_error: true
+          action: codecov/codecov-action@v3
```
Comment (meskill) [inline .github/workflows/ci.yml:81]: return back to `v4` =)

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -76,11 +76,15 @@ jobs:
...
+          attempt_limit: 10
```
Comment (meskill) [inline .github/workflows/ci.yml:82]: let's have like 3 attempts

**Review by meskill — DISMISSED** [review]

File: .github/workflows/ci.yml (hunk 3 above)
Comment (ologbonowiwi) [reply .github/workflows/ci.yml]: changed, but 3 attempts likely will not solve the issue 100%, as usually codecov stays down for a couple of seconds/minutes.

**Review by tusharmath — APPROVED** [review]

**Pull request merged**